py7zr/archiveinfo.py

```
@@ -1117,14 +1117,22 @@ def _read(self, file: BinaryIO) -> None:
+            self.startheadercrc, _ = read_uint32(file)
+            self.nextheaderofs, data = read_real_uint64(file)
+            crc = calculate_crc32(data)
+            self.nextheadersize, data = read_real_uint64(file)
+            crc = calculate_crc32(data, crc)
+            self.nextheadercrc, data = read_uint32(file)
+            crc = calculate_crc32(data, crc)
+        except struct.error:
+            # A file that starts with the 7z magic but is cut off inside the
+            # 32-byte signature header would otherwise surface a raw
+            # struct.error from unpack() on the short buffer.
+            raise Bad7zFile("truncated 7z signature header")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)